gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb

```diff
@@ -0,0 +1,484 @@
+        "OPTIONS (endpoint = 'text-embedding-004')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JcYVG4qNKAuY"
+      },
+      "source": [
+        "# Prepare the dataset for semantic search"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KCZ-nme8GjTN"
+      },
+      "source": [
+        "## Generate text embeddings for title and body associated with Stack Overflow questions"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1VPgFMlBGi_5"
+      },
+      "outputs": [],
+      "source": [
+        "%%bigquery\n",
+        "CREATE OR REPLACE TABLE\n",
+        "  `bigquery_demo.posts_questions_embedding` AS\n",
+        "SELECT\n",
+        "  *\n",
+        "FROM\n",
+        "  ML.GENERATE_EMBEDDING( MODEL `bigquery_demo.text_embedding_004`,\n",
+        "    (\n",
+        "    SELECT\n",
+        "      id,\n",
+        "      title,\n",
+        "      body,\n",
+        "      CONCAT (title, body ) AS CONTENT\n",
+        "    FROM\n",
+        "      `bigquery-public-data.stackoverflow.posts_questions`\n",
+        "    LIMIT\n",
+        "      10000 ),\n",
+        "    STRUCT ( TRUE AS flatten_json_output,\n",
+        "      'SEMANTIC_SIMILARITY' AS task_type ) );"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8bcyGsNWHPeZ"
+      },
+      "outputs": [],
+      "source": [
+        "%%bigquery\n",
+        "SELECT * FROM `bigquery_demo.posts_questions_embedding` LIMIT 100;"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ceUN_Mgm0vQa"
+      },
+      "source": [
+        "## Create Vector Index on the embeddings to help with efficient semantic search"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Gi1WMtnp1Tvh"
+      },
+      "outputs": [],
+      "source": [
+        "%%bigquery\n",
+        "  CREATE OR REPLACE VECTOR INDEX ix_posts_questions\n",
+        "  ON\n",
+        "  `bigquery_demo.posts_questions_embedding` (ml_generate_embedding_result) OPTIONS(index_type = 'IVF',\n",
+        "    distance_type = 'COSINE',\n",
+        "    ivf_options = '{\"num_lists\":500}');"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "M4swkWSH08yE"
+      },
+      "source": [
+        "## Verify vector index creation"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "v-MJ0Gtui_BZ"
+      },
+      "source": [
+        "Note : The vector index is populated asynchronously. You can check whether the index is ready to be used by querying the INFORMATION_SCHEMA.VECTOR_INDEXES view and verifying that the coverage_percentage column value is greater than 0 and the last_refresh_time column value isn't NULL."
```

Comment:
```suggestion
        "Note : The vector index is populated asynchronously. You can check whether the index is ready to be used by querying the `INFORMATION_SCHEMA.VECTOR_INDEXES` view and verifying that the `coverage_percentage` column value is greater than 0 and the `last_refresh_time` column value isn't `NULL`."
```

Do similar for any "code-like" value in markdown

Reply:
Done